### ISSUE TITLE
Error log entries due to zipdownload

### ISSUE THREAD

**Issue by lmaylein** [IQSS/dataverse#7693]
The error log of our Apache server has been flooded with messages of this kind for some time. I suspect this is caused by zipdownload. Can these messages be disabled?

[Wed Mar 17 13:44:50.619055 2021] [cgi:error] [pid 17800] [client 129.206.xxx.xxx:35826] AH01215: offset: 0, length: 8192, referer: https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/HVXXIJ
[Wed Mar 17 13:44:50.621371 2021] [cgi:error] [pid 17800] [client 129.206.xxx.xxx:35826] AH01215: offset: 0, length: 8192, referer: https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/HVXXIJ
[Wed Mar 17 13:44:50.621652 2021] [cgi:error] [pid 17800] [client 129.206.xxx.xxx:35826] AH01215: offset: 0, length: 8192, referer: https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/HVXXIJ

**Comment by djbrooke** [comment]
Hey @lmaylein - we'll check to see if we're seeing the same on dataverse.harvard.edu. Apologies if this has already been asked, but are you running on S3 or other storage?

**Comment by lmaylein** [comment]
local file(system) storage

**Comment by landreev** [comment]
Hello, 
The easiest way to disable these messages is to change this line in your /etc/httpd/conf.d/ssl.conf:
```
LogLevel warn
```
to 
```
LogLevel crit
```

I'm assuming these messages are NOT an indication of a serious problem. My guess is that this is being caused by some crawler bots finding the zipper url in the dataverse pages and generating bad requests trying to follow it?  Of course they can't download anything, without Dataverse generating a proper download authorization in the database table.
Let me try and figure out for sure what may be causing it.

**Comment by landreev** [comment]
P.S. If you could match these messages (for example, the 3 messages above) in your error log, to the exact requests in your `/var/log/httpd/ssl_access_log*`, that would help. 
Thank you.

**Comment by lmaylein** [comment]
> P.S. If you could match these messages (for example, the 3 messages above) in your error log, to the exact requests in your /var/log/httpd/ssl_access_log*, that would help.

It's not quite that simple. The timestamps in access_log differ significantly from those of the presumably associated entries in error_log. The surrounding log file entries in access_log also indicate that the download requests ran for an extremely long time (many hours), presumably because the requesting client only accepted the data very slowly. Perhaps this is also the cause of these error log entries? I suspected that zipdownload was writing some debug information to STDERR.

[Wed Mar 17 22:33:08.223755 2021] [cgi:error] [pid 28881] [client 111.27.xxx.xxx:9654] AH01215: offset: 0, length: 8192, referer: https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/V4MUMX
[Wed Mar 17 22:33:08.225030 2021] [cgi:error] [pid 28881] [client 111.27.xxx.xxx:9654] AH01215: offset: 0, length: 8192, referer: https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/V4MUMX
[Wed Mar 17 22:33:08.225073 2021] [cgi:error] [pid 28881] [client 111.27.xxx.xxx:9654] AH01215: offset: 0, length: 8192, referer: https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/V4MUMX


111.27.xxx.xxx - - [17/Mar/2021:17:13:09 +0100] "POST /dataset.xhtml?persistentId=doi:10.11588/data/V4MUMX HTTP/1.1" 200 212 "https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/V4MUMX" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.108 Safari/537.36"
111.27.xxx.xxx - - [17/Mar/2021:17:13:10 +0100] "GET /cgi-bin/zipdownload?bc6-267a5f7e369c HTTP/1.1" 200 274000 "https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/V4MUMX" "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.108 Safari/537.36"
111.27.xxx.xxx - - [17/Mar/2021:17:13:39 +0100] "GET /cgi-bin/zipdownload?bc6-267a5f7e369c HTTP/1.1" 200 1206302000 "https://heidata.uni-heidelberg.de/dataset.xhtml?persistentId=doi:10.11588/data/V4MUMX" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36"

**Comment by landreev** [comment]
OK, I see. There are multiple log messages within the same millisecond... So they are clearly coming from the same download job. Wondering if this starts happening if the remote client times out/otherwise completely stops accepting the output? 
Apparently we had a few episodes of this happening early on; and then somebody from our systems support group simply changed the Apache config to stop logging anything below "critical", and it never got investigated. 
I still think it's fairly safe to simply block these messages. But I'll try to figure out what's going on.

**Comment by pdurbin** [comment]
@lmaylein are you still suffering from this problem?

**Comment by lmaylein** [comment]
> are you still suffering from this problem?
No, but that may also have something to do with the fact that we have now set the limit for zip downloads (ZipDownloadLimit) to 5 GByte.

**Comment by pdurbin** [comment]
@lmaylein hmm, ok, so there still might be a bug in there. Thanks!

If anyone out there can reproduce this or is thinking about making a pull request, please get in touch!
